certora/harness/AbstractBaseHarness.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {IEVC} from "ethereum-vault-connector/interfaces/IEthereumVaultConnector.sol";
+import "../../src/EVault/shared/Base.sol";
+// Needed for checkLiquidityReturning:
+import {LiquidityUtils} from "../../src/EVault/shared/LiquidityUtils.sol";
+
+// This exists so that Base.LTVConfig and other type declarations 
+// are available in CVL and can be used across specs for different modules.
+// We need to split this into a concrete contract and an Abstract contract
+// so that we can refer to Base.LTVConfig as a type in shared CVL functions
+// while also making function definitions sharable among harnesses via
+// AbstractBase. AbstractBaseHarness includes the shared function definitions.
+abstract contract AbstractBaseHarness is Base, LiquidityUtils {
+
+    function getLTVConfig(address collateral) external view returns (LTVConfig memory) {
+        return vaultStorage.ltvLookup[collateral];
+    }
+
+    function vaultCacheOracleConfigured() external returns (bool) {
+        return address(loadVault().oracle) != address(0);
+    }
+
+    function isAccountStatusCheckDeferredExt(address account) external view returns (bool) {
+        return isAccountStatusCheckDeferred(account);
+    }
+    
+    function getBalanceAndForwarderExt(address account) public returns (Shares, bool) {
+        return vaultStorage.users[account].getBalanceAndBalanceForwarder();
+    }
+
+    function checkAccountMagicValue() public view returns (bytes4) {
+        return IEVCVault.checkAccountStatus.selector;
+    }
+
+    function checkAccountMagicValueMemory() public view returns (bytes memory) {
+        return abi.encodeWithSelector(IEVCVault.checkAccountStatus.selector);
+    }
+
+    function checkVaultMagicValueMemory() public view returns (bytes memory) {
+        return abi.encodeWithSelector(IEVCVault.checkVaultStatus.selector);
+    }
+
+    function getUserInterestAccumulator(address account) public view returns (uint256) {
+        return vaultStorage.users[account].interestAccumulator;
+    }
+
+    // This mirrors LiquidityUtils.checkLiquidity except that it returns
+    // a bool rather than reverting.
+    function checkLiquidityReturning(address account, address[] memory collaterals) public returns (bool) {
```

Comment:
can we have a rule that shows that this function returns false iff checkLiquidity reverts

Reply:
This is another one that could be looked at with Maya's team, potentially (see comment here: https://github.com/euler-xyz/euler-vault-kit/pull/256#discussion_r1707759770)

Reply:
I'm working on this currently.

Reply:
OK we proved this property as well. It's called `checkLiquidityReturningSameAsOriginal`. It's in `RiskManager.spec`.

https://prover.certora.com/output/40726/f67c06400ebc412c88740b7efe675bc4/?anonymousKey=7101510a9ac7abec41dab0ee4f659219da5b98b9